pkg/apis/pipeline/v1/param_types.go

```diff
@@ -133,8 +146,8 @@ func (ps ParamSpecs) sortByType() (ParamSpecs, ParamSpecs, ParamSpecs) {
 // validateNoDuplicateNames returns an error if any of the params have the same name
 func (ps ParamSpecs) validateNoDuplicateNames() *apis.FieldError {
 	names := ps.getNames()
-	seen := sets.String{}
-	dups := sets.String{}
+	seen := sets.New[string]()
+	dups := sets.New[string]()
```

Comment:
I started this and then stopped here. `sets.Strings{}` are deprecated, and they're easy to replace with the new generics-based ones, but changes to one method have many ramifications in the validation code base, so that's a change we might want to do as a separate PR or set of PRs.

Reply:
I believe there was an issue opened by Vincent on this https://github.com/tektoncd/pipeline/issues/7374.